src/types/ajv/CycleRecordSchema.ts

```diff
@@ -0,0 +1,284 @@
+                  },
+                  stake: {
+                    type: 'object',
+                    properties: {
+                      dataType: { type: 'string' },
+                      value: { type: 'string' },
+                    },
+                    required: ['dataType', 'value'],
+                  },
+                },
+                required: ['certExp', 'nominator', 'nominee', 'signs', 'stake'],
+              },
+              version: { type: 'string' },
+            },
+            required: ['mustUseAdminCert', 'stakeCert', 'version'],
+          },
+          cycleMarker: { type: 'string' },
+          nodeInfo: {
+            type: 'object',
+            properties: {
+              publicKey: { type: 'string' },
+              externalIp: { type: 'string' },
+              externalPort: { type: 'number' },
+              internalIp: { type: 'string' },
+              internalPort: { type: 'number' },
+              joinRequestTimestamp: { type: 'number' },
+              readyTimestamp: { type: 'number' },
+              address: { type: 'string' },
+            },
+            required: [
+              'publicKey',
+              'externalIp',
+              'externalPort',
+              'internalIp',
+              'internalPort',
+              'joinRequestTimestamp',
+              'address',
+            ],
+          },
+          proofOfWork: { type: 'string' },
+          selectionNum: { type: 'string' },
+          sign: {
+            type: 'object',
+            properties: {
+              owner: { type: 'string' },
+              sig: { type: 'string' },
+            },
+            required: ['owner', 'sig'],
+          },
+          version: { type: 'string' },
+        },
+        required: [
+          'appJoinData',
+          'cycleMarker',
+          'nodeInfo',
+          'proofOfWork',
+          'selectionNum',
+          'sign',
+          'version',
+        ],
+      },
+    },
+    standbyNodeListHash: { type: 'string' },
+    standbyRefresh: { type: 'array', items: { type: 'string' } },
+    standbyRemove: { type: 'array', items: { type: 'string' } },
+    startedSyncing: { type: 'array', items: { type: 'string' } },
+    syncing: { type: 'number' },
+    target: { type: 'number' },
+    random: { type: 'number' },
+    networkDataHash: {
+      type: 'array',
+      items: {
+        type: 'object',
+        properties: {
+          cycle: { type: 'number' },
+          hash: { type: 'string' },
+        },
+      },
+    },
+    networkReceiptHash: {
+      type: 'array',
+      items: {
+        type: 'object',
+        properties: {
+          cycle: { type: 'number' },
+          hash: { type: 'string' },
+        },
+      },
+    },
+    networkSummaryHash: {
+      type: 'array',
+      items: {
+        type: 'object',
+        properties: {
+          cycle: { type: 'number' },
+          hash: { type: 'string' },
+        },
+      },
+    },
+  },
+  required: [
```

Comment:
it seems like there should be more items in the required list here?  are the missing fields actually safe to ignore.
Also AJV should have a setting to reject if there are extra fields getting injected

Reply:
I can add more fields that should in theory never be null to the required fields however my concern is that there’s edge cases that do have null fields and since sync of cycle records is pretty central I think that’d be a bit risky. current validation isn’t checking for other fields than the ones I included in the schema either. 
Should we wait for the RED-300 task before adding more required fields (just to be sure)?

Reply:
added all the fields that should in theory be there now. if this fails us we can turn off the AJV validation flag still